### ISSUE TITLE
Cannot use custom templates with swagger-codegen-v3

### ISSUE THREAD

**Issue by OnkelTem** [swagger-api/swagger-codegen#10107]
Hi!

I followed the [FAQ](https://github.com/swagger-api/swagger-codegen/wiki/FAQ#how-can-i-customize-the-template-to-add-headerfooter-to-the-auto-generated-code) to customize one template from `nodejs-server` generator.

According to the FAQ I copied just one file from this repo - `modules/swagger-codegen/src/main/resources/nodejs/controller.mustache` - to my custom directory which I mount into docker image - [swaggerapi/swagger-codegen-cli-v3](https://hub.docker.com/r/swaggerapi/swagger-codegen-cli-v3) (of which I couldn't find any information - neither on that Docker Hub page, nor in this repo - anywhere. So I just guessed it's been sourced from the same `modules/swagger-codegen-cli` dir which seems to be wrong assumption!)

Then I ran it like with `-t` set to my custom directory and received this error message:

```
Generate models: Exception in thread "Thread-1" 
Generate models: java.lang.RuntimeException: Could not generate api file for 'Auth'

Generate models:        at io.swagger.codegen.v3.DefaultGenerator.generateApis(DefaultGenerator.java:579)

Generate models:        at io.swagger.codegen.v3.DefaultGenerator.generate(DefaultGenerator.java:782)
        at io.swagger.codegen.v3.cli.cmd.Generate.run(Generate.java:360)
        at java.lang.Thread.run(Thread.java:748)

Generate models: Caused by: java.io.FileNotFoundException: handlebars/nodejs/service.mustache

Generate models:        at com.github.jknack.handlebars.io.URLTemplateLoader.sourceAt(URLTemplateLoader.java:70)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:438)

Generate models:        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:419)
        at io.swagger.codegen.v3.templates.HandlebarTemplateEngine.getHandlebars(HandlebarTemplateEngine.java:48)
Generate models: 
        at io.swagger.codegen.v3.templates.HandlebarTemplateEngine.getRendered(HandlebarTemplateEngine.java:24)

Generate models:        at io.swagger.codegen.v3.DefaultGenerator.processTemplateToFile(DefaultGenerator.java:795)
        at io.swagger.codegen.v3.DefaultGenerator.generateApis(DefaultGenerator.java:538)
Generate models: 
        ... 3 more
Generate models:
```

So it was searching for some `handlebars/nodejs/service.mustache` file which is definitely not here or there: first off, there is not `handlebars` dir...

Do you have any ideas how to do it at last?

**Comment by OnkelTem** [comment]
Ron has helped me a little with locating the source of the docker image, and so finally I could make a more cleaner test. TL;DR: it fails too.

1. I downloaded the latest JAR file using this command from the README:
```
$ wget https://repo1.maven.org/maven2/io/swagger/codegen/v3/swagger-codegen-cli/3.0.18/swagger-codegen-cli-3.0.18.jar -O swagger-codegen-cli.jar

```
2. Then I cloned https://github.com/swagger-api/swagger-codegen-generators repo and copied `handlebars` directory under locally created `./generator` directory.

3. Then I ran the generation using this command line:
```
$ java -jar swagger-codegen-cli.jar generate -i swagger/tsanalyzer.yaml -l nodejs-server -t generator -o src/
```
and it failed with this error:
```
Exception in thread "Thread-1" java.lang.RuntimeException: Could not generate api file for 'Auth'
        at io.swagger.codegen.v3.DefaultGenerator.generateApis(DefaultGenerator.java:579)
        at io.swagger.codegen.v3.DefaultGenerator.generate(DefaultGenerator.java:782)
        at io.swagger.codegen.v3.cli.cmd.Generate.run(Generate.java:360)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.FileNotFoundException: handlebars/nodejs/service.mustache
        at com.github.jknack.handlebars.io.URLTemplateLoader.sourceAt(URLTemplateLoader.java:70)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:438)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:419)
        at io.swagger.codegen.v3.templates.HandlebarTemplateEngine.getHandlebars(HandlebarTemplateEngine.java:48)
        at io.swagger.codegen.v3.templates.HandlebarTemplateEngine.getRendered(HandlebarTemplateEngine.java:24)
        at io.swagger.codegen.v3.DefaultGenerator.processTemplateToFile(DefaultGenerator.java:795)
        at io.swagger.codegen.v3.DefaultGenerator.generateApis(DefaultGenerator.java:538)
        ... 3 more
```
So basically, as you see it cannot find `handlebars/nodejs/service.mustache` file, which is acutally there, under `./generator/handlebars/nodejs/service.mustache`.

**So how can I override a template?**

Thanks in advance.

**Comment by OnkelTem** [comment]
I tried passing relative path:

```
java -jar swagger-codegen-cli.jar generate -i swagger-api/tsanalyzer.yaml -l nodejs-server -t ./generator -o src/
```
absolute path:
```
java -jar swagger-codegen-cli.jar generate -i swagger-api/tsanalyzer.yaml -l nodejs-server -t /projects/streamlabs/tsanalyzer/mock-server/generator -o src/
```
I tried to copy the `nodejs` dir under `./generator/`:
```
generator
  nodejs
    controller.mustache
    index.mustache
    index-gcf.mustache
    package.mustache
    README.mustache
    service.mustache
    swagger.mustache
    writer.mustache
```
as well as the content of the `nodejs` dir in there:
```
generator
  controller.mustache
  index.mustache
  index-gcf.mustache
  package.mustache
  README.mustache
  service.mustache
  swagger.mustache
  writer.mustache
```
but NOTHING seems to work! It's still complaining about FileNotFoundException:
```
Exception in thread "Thread-1" java.lang.RuntimeException: Could not generate api file for 'Auth'
        at io.swagger.codegen.v3.DefaultGenerator.generateApis(DefaultGenerator.java:579)
        at io.swagger.codegen.v3.DefaultGenerator.generate(DefaultGenerator.java:782)
        at io.swagger.codegen.v3.cli.cmd.Generate.run(Generate.java:360)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.FileNotFoundException: handlebars/nodejs/service.mustache
        at com.github.jknack.handlebars.io.URLTemplateLoader.sourceAt(URLTemplateLoader.java:70)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:438)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:419)
        at io.swagger.codegen.v3.templates.HandlebarTemplateEngine.getHandlebars(HandlebarTemplateEngine.java:48)
        at io.swagger.codegen.v3.templates.HandlebarTemplateEngine.getRendered(HandlebarTemplateEngine.java:24)
        at io.swagger.codegen.v3.DefaultGenerator.processTemplateToFile(DefaultGenerator.java:795)
        at io.swagger.codegen.v3.DefaultGenerator.generateApis(DefaultGenerator.java:538)
        ... 3 more
```

Hello?

**Comment by HugoMario** [comment]
thanks for reporting this, going to assign myself and work asap on it

**Comment by HugoMario** [comment]
this has been fixed by [swagger-codegen-generators#644](https://github.com/swagger-api/swagger-codegen-generators/pull/644)

**Comment by lmvdev** [comment]
Also, fix for Swift5, please

**Comment by georgedias** [comment]
Having the same issue with Ruby.
Can this be fixed for ruby as well, thanks.
